### PR TITLE
Configure podman to trust katello CA on containerized Satellite

### PR DESCRIPTION
**Pull request by vsedmik**
### Problem Statement
Podman on foremanctl-deployed (containerized) Satellites doesn't trust the katello server CA by default. This causes TLS verification failures (x509: certificate signed by unknown authority) when tests use podman push/podman pull against the Satellite's own container registry.

### Solution
Add a session-scoped autouse fixture `configure_podman_ca_trust` in `pytest_fixtures/core/sys.py` that copies the katello CA cert into `/etc/containers/certs.d/<hostname>/` on containerized Satellites, following the upstream documentation. The fixture only runs for foremanctl-installed Satellites and reuses existing helpers (ca_cert_file, CONTAINER_CERTS_PATH).

### Related Issues
No issues

### PRT test Cases example
```
trigger: test-robottelo
pytest: tests/foreman/api/test_docker.py::TestPodman -k test_cv_podman
```

## Summary by Sourcery

New Features:
- Add a session-scoped autouse fixture that sets up podman CA trust for containerized Satellites using the katello CA certificate.

### CODE REVIEW

**Comment by vsedmik** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/api/test_docker.py::TestPodman -k test_cv_podman
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 16386
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/api/test_docker.py::TestPodman -k test_cv_podman --external-logging
Test Result : =========== 1 passed, 3 deselected, 6 warnings in 1074.95s (0:17:54) ===========
```

**Review by aidenfine — APPROVED** [review]
LGTM

File: pytest_fixtures/core/sys.py [hunk 1]
```diff
@@ -51,6 +53,16 @@ def puppet_proxy_port_range(session_puppet_enabled_sat):
             )
 
 
+@pytest.fixture(autouse=True, scope='session')
+def configure_podman_ca_trust(session_target_sat):
+    """Configure podman on containerized Satellite to trust its own katello CA."""
+    if session_target_sat and session_target_sat.install_method == InstallMethod.FOREMANCTL:
```
Comment (Gauravtalreja1) [inline pytest_fixtures/core/sys.py:59]: I'd say checking the setting would be more effective here
```suggestion
    if settings.server.install_method == InstallMethod.FOREMANCTL:
```

File: pytest_fixtures/core/sys.py [hunk 2]
```diff
@@ -51,6 +53,16 @@ def puppet_proxy_port_range(session_puppet_enabled_sat):
...
+        certs_dir = f'{CONTAINER_CERTS_PATH}{session_target_sat.hostname}'
+        session_target_sat.execute(
+            f'mkdir -p {certs_dir} && cp {session_target_sat.ca_cert_file} {certs_dir}/ca.crt'
```
Comment (Gauravtalreja1) [inline pytest_fixtures/core/sys.py:62]: I'm not sure if we should workaround here for our tests, and shouldn't this be part of the foremanctl deployment itself? It seems like a natural fit for the certificates or post_install role in foremanctl, that way any container content workflows work out of the box without CU having to set it up themselves.

wdyt @evgeni @ekohl @ehelms ?

Comment (ekohl) [reply pytest_fixtures/core/sys.py:62]: We used to do that in our classic installer, but we don't in foremanctl. IMHO we shouldn't blindly trust it and I'd also be hesitant to implement this in robottelo. It means the test you're running doesn't reflect the real world that customers run.

AFAIK customers shouldn't be pulling podman containers from their Satellite. That is the self-managed concept and we don't support it.

@vsedmik can you explain why you think this is needed? What is it solving?

Comment (vsedmik) [reply pytest_fixtures/core/sys.py:62]: This is supposed as a quick fix for tests that use Satellite itself as a podman client, which we have been using since long.

That said, I'm not 100% convinced we should keep doing that, but the proper "customer workflow" alternative (spin up, configure, register RHEL host via AK to an CVEnv) has its price we should consider - increased execution time and more VMs deployed.

I'm curious what @ehelms thinks.

Comment (evgeni) [reply pytest_fixtures/core/sys.py:62]: I would prefer if we test what customers deploy. So a separate host, global registration, podman login/pull.

Comment (ehelms) [reply pytest_fixtures/core/sys.py:62]: I agree with what @evgeni said.

Comment (synkd) [reply pytest_fixtures/core/sys.py:62]: For my own edification, where do we document that the Foreman/Satellite server should not be used to pull/push container images to its own registry? It's not clear to me from e.g. the content management Guide [1] that this is the case.

[1] https://docs.redhat.com/en/documentation/red_hat_satellite/6.19/html-single/managing_content/index#Using_Container_Registries_content-management

Comment (evgeni) [reply pytest_fixtures/core/sys.py:62]: We do not (even tho I'd really like us to discourage the user from doing so).

Comment (ekohl) [reply pytest_fixtures/core/sys.py:62]: > For my own edification, where do we document that the Foreman/Satellite server should not be used to pull/push container images to its own registry?

https://docs.redhat.com/en/documentation/red_hat_satellite/6.19/html/installing_satellite_server_in_a_connected_network_environment/installing-satellite-server states
> You cannot register Satellite Server to itself.

You can argue how we exactly determine that and pulling containers is a bit of an edge case.

If you pull images on the host itself, I would do it at the user level and not as root. You can even create a random non-root user for the specific test. However, that is a big if and I'd agree with what @evgeni said: separate host is better.

**Comment by vsedmik** [comment]
Agreed on fixing the issue "the proper way" with registered content host.
